src/main/java/com/epam/ta/reportportal/dao/TestItemRepository.java

```diff
@@ -33,6 +39,62 @@
 	@Query(value = "SELECT ti.id FROM TestItem ti WHERE ti.launch.id = :launchId")
 	Stream<Long> streamTestItemIdsByLaunchId(@Param("launchId") Long launchId);
 
+	/**
+	 * Retrieve the {@link List} of the {@link TestItem#itemId} by launch ID, {@link StatusEnum#name()} and {@link TestItem#hasChildren} == false
+	 *
+	 * @param launchId {@link com.epam.ta.reportportal.entity.launch.Launch#id}
+	 * @param status   {@link StatusEnum#name()}
+	 * @return the {@link List} of the {@link TestItem#itemId}
+	 */
+	@QueryHints(value = @QueryHint(name = HINT_FETCH_SIZE, value = "1"))
+	@Query(value = "SELECT test_item.item_id FROM test_item JOIN test_item_results result on test_item.item_id = result.result_id "
+			+ " WHERE test_item.launch_id = :launchId AND NOT test_item.has_children AND result.status = cast(:status as status_enum)", nativeQuery = true)
+	Stream<BigInteger> streamIdsByNotHasChildrenAndLaunchIdAndStatus(@Param("launchId") Long launchId, @Param("status") String status);
```

Comment:
Why BigInteger?

Reply:
In our DB ids are BIGINT, that's why their representation in code is also a BigInteger, and also it's a native query, it means that BIGINT won't be casted to Long implicitly, so the only way is to get it as BigInteger and then call 'longValue()' method.